astropy/convolution/setup_package.py

```diff
@@ -12,8 +13,15 @@
 if not sys.platform.startswith("win"):
     extra_compile_args.append("-fPIC")
 
+USE_PY_LIMITED_API = not sysconfig.get_config_var("Py_GIL_DISABLED")
```

Comment:
I have a slight preference for making this option opt-in, namely by making it conditional on an env var (`ASTROPY_LIMITED_API` ?), because consumers who build from source generally don't need the portability, and should optimize for performance instead.

Reply:
and it's easy enough to set such an env var within our cibuildwheel config, that won't affect downstream consumers

Reply:
We can make this opt-in for now since we aren't actually tagging the wheel abi3, but just to check, how do you envisage this working once we add the ``py-limited-api`` option to ``pyproject.toml``? (in the sense that either the option is there or it isn't, it's difficult to control conditionally)

Reply:
Why does it need to be declared statistically ? None of the setuptools-based projects I worked on had this constraint.

Reply:
Ah, how do they declare it?

Reply:
Dynamically, by subclassing `wheel.bdist_wheel`

See the example I linked in https://github.com/astropy/astropy/pull/18211#discussion_r2129167635